pkg/controllers/managementuser/rbac/globalrolebinding_handler.go

```diff
@@ -179,10 +182,32 @@ func (c *grbHandler) ensureProvisioningClusterAdminBinding(obj *v3.GlobalRoleBin
 	}
 
 	provCluster := pClusters[0]
-
 	subject := rbac.GetGRBSubject(obj)
-	rbName := name.SafeConcatName(rbac.ProvisioningClusterAdminName(provCluster), rbac.GetGRBTargetKey(obj))
 
+	// Add the restricted admin user as a member of the downstream cluster
+	// by creating a CRTB in the local custer in the namespace named after the downstream cluster.
+	_, err = c.clusterRoleTemplateBindings.Create(&v3.ClusterRoleTemplateBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("crtb-%s-%s-%s", subject.Name, "restricted-admin", "cluster-owner"),
+			Namespace: provCluster.Name,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: obj.APIVersion,
+					Kind:       obj.Kind,
+					Name:       obj.Name,
+					UID:        obj.UID,
+				},
+			},
+		},
+		ClusterName:      provCluster.Name,
+		UserName:         subject.Name,
```

Comment:
The subject can either be a username or a groupPrincipal name so we need to check and create the CRTB accordingly.

Reply:
We already do, since the `subject` is derived from `subject := rbac.GetGRBSubject(obj)`. That function handles groups and users.

Reply:
Should I set UserName or UserPrincipalName? Likewise, GroupName or GroupPrincipalName?
This is derived from the Subject, which has one (of either).